improver/utilities/gradient_between_vertical_levels.py

```diff
@@ -99,8 +102,10 @@ def gradient_over_vertical_levels(
             else:
                 if orography:
                     height_ASL = orography + cube_height
+                    coord_list.append("pressure")
                 elif not (orography or geopotential_height):
                     height_ASL = cube_height
+                    coord_list.append("pressure")
```

Comment:
This is confusing. If there was a pressure coord on either input cube, `coord_list` contains `"height"` and if there was a height coord on either input cube, `coord_list` contains `"pressure"`. The logic all works because it depends on both being present. I suggest reversing the logic and naming the variable `coords_used`.

Reply:
I've reversed the logic. I think it was originally like that because I was using the coord names for something but I don't remember what.